versioned-docs.md

```diff
@@ -0,0 +1,23 @@
+# Versioned Documentation
+
+We are starting to run into issues with people trying to use Fairlearn installed from PyPI with the documentation on [our website](https://fairlearn.github.io/).
+This is an issue because we have made and are making a number of breaking changes to our API.
+The solution to this is to provide versioned documentation, so that users can read the docs associated with the version they have installed.
+This is the pattern used by projects such as [SciKit-Learn](https://scikit-learn.org).
+
+Getting to something as sophisticated as the SciKit-Learn setup will take time.
+As a first step, we propose:
+
+1. Change the CirceCI integration to push to a `main` directory (in preparation for the expected rename of the `master` branch) within the [`fairlearn.github.io`](https://github.com/fairlearn/fairlearn.github.io) repository, rather than pushing to the root of the repository.
+
+1. Update links on the custom-designed landing page to go to this new `main` directory.
+
+1. Create a `v0.4.6` directory in `fairlearn.github.io`, and do a documentation build corresponding to the `v0.4.6` tag in this directory. This may well have to be done manually.
+
+The last stage will at least give us a set of documentation corresponding to the version of Fairlearn on PyPI, which we can share with people who run into difficulties.
+
```

Comment:
Any idea about the tools you want to use for this? @cedarfall looked into some of them recently and found:
- https://stackoverflow.com/questions/60487622/multiple-versions-in-self-hosted-sphinx-documentation
- https://github.com/sphinx-contrib/sphinxcontrib-versioning
- https://pypi.org/project/sphinx-multiversion/
- https://sphinxcontrib-versioning.readthedocs.io/en/latest/tutorial.html

Reply:
Will do some more reading. When I was writing this, I was primarily thinking about the v0.4.6 issue (not to mention the even older versions for which all the sphinx docs will vanish).

Reply:
I don't think we need to care. Scikit learn starts somewhere around 0.17 or so and that's fine.

Reply:
I didn't mean to go back to the beginning, but I would like to be able to generate docs for v0.4.6.

I've been reading the docs for both sphinxcontrib-versioning and sphinx-multiversion. Both look like they can go back to older versions just fine.

Comparing:
https://github.com/sphinx-contrib/sphinxcontrib-versioning
and
https://github.com/Holzhaus/sphinx-multiversion
I'm a little concerned that the former hasn't had any activity since last December, and it was mostly 'done' back in 2016. I'll start by seeing that sphinx-multiversion can do.

Reply:
Been trying to persuade `sphinx-multiversion` to work... have wound up filing an issue on it :-/